cmd/thanos/tools_bucket.go

```diff
@@ -440,6 +443,18 @@ func registerBucketLs(app extkingpin.AppClause, objStoreConfig *extflag.PathOrCo
 			return err
 		}
 
+		if tbc.excludeDelete {
+			// Temporary placeholder values - should I add deleteDelay as a parameter for Ls too?
+			f := block.NewIgnoreDeletionMarkFilter(logger, bkt, 0, 1)
+
+			// Created fetcherMetrics since Filter required an *extprom.TxGaugeVec.
+			fetcherMetrics := block.NewFetcherMetrics(reg, nil, nil)
```

Comment:
The filtering step can be added to line 383

Reply:
Thanks @yeya24 ! This approach is definitely better. Fixed in a recent commit.